src/Simplify.elm

```diff
@@ -2876,6 +2911,154 @@ stringReverseChecks ({ parentRange, fnRange, firstArg } as checkInfo) =
+                    , end = (Node.range checkInfo.firstArg).end
+                    }
+                    "String.left"
+                ]
+            ]
+
+        ( start, Just end, _ ) ->
+            Maybe.map2
+                (\startInt endInt ->
+                    if
+                        (startInt >= endInt)
+                            && -- have the same sign
+                               ((startInt <= -1 && endInt <= -1)
+                                    || (startInt >= 0 && endInt >= 0)
+                               )
+                    then
+                        [ Rule.errorWithFix
+                            { message = "The call to String.slice will result in \"\""
+                            , details = [ "You can replace this slice operation by \"\"." ]
+                            }
+                            checkInfo.fnRange
+                            [ Fix.replaceRangeBy checkInfo.parentRange "always \"\"" ]
+                        ]
+                            |> Just
+
+                    else
+                        -- either is negative or startInt < endInt
+                        Nothing
+                )
+                (Evaluate.getInt checkInfo start)
+                (Evaluate.getInt checkInfo end)
+                |> Maybe.withDefault
+                    (if Normalize.areAllTheSame checkInfo start [ end ] then
+                        [ Rule.errorWithFix
+                            { message = "Using String.slice with equal start and end index will result in an empty string"
+                            , details = [ "You can replace this call by an empty string." ]
+                            }
+                            checkInfo.fnRange
+                            [ Fix.replaceRangeBy checkInfo.parentRange "always \"\"" ]
+                        ]
+                            |> Just
+
+                     else
+                        Nothing
+                    )
+                |> Maybe.withDefault []
+
+        ( _, Nothing, _ ) ->
+            []
+
+
+stringLeftChecks : CheckInfo -> List (Error {})
+stringLeftChecks checkInfo =
+    case ( checkInfo.firstArg, checkInfo.secondArg ) of
+        ( _, Just (Node _ (Expression.Literal "")) ) ->
+            [ Rule.errorWithFix
+                { message = "Using String.left on an empty string will result in an empty string"
+                , details = [ "You can replace this call by an empty string." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange "\"\"" ]
+            ]
+
+        ( Node _ (Expression.Integer 0), _ ) ->
+            [ Rule.errorWithFix
+                { message = "Using String.left with length 0 will result in an empty string"
+                , details = [ "You can replace this call by an empty string." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange "always \"\"" ]
```

Comment:
This should probably be `""` if there is a second argument, and ``always ""`` if there isn't.